chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -48,6 +54,12 @@ def forward_cpu(self, inputs):
         log_y = log_softmax._log_softmax(x, self.use_cudnn)
         if self.cache_score:
             self.y = numpy.exp(log_y)
+        if self.class_weight is not None:
+            if self.class_weight.shape != x.shape:
+                shape = [1 if d != 1 else -1 for d in six.moves.range(x.ndim)]
+                self.class_weight = numpy.broadcast_to(
+                    self.class_weight.reshape(shape), x.shape)
+            log_y *= self.class_weight
```

Comment:
It requires the same computation (calculate `shape` and broadcast `class_weight` to `x.shape`) again in backward_cpu, so it leads a duplicate computation. So how about making `class_weight` private member like `_class_weight`?